docs/content/intro/tutorial-preview-environments.md

```diff
@@ -0,0 +1,424 @@
+? Clone URL https://github.com/brigadecore/brigade-tutorial-app
+Auto-generated a Shared Secret: "uSEtlJicRK3RhRWiOatImwBs"
+? Configure GitHub Access? Yes
+? OAuth2 token <my-github-token>
+```
+
+## GitHub Webhook
+
+To enable auto-deployment of our service to all (interested) preview environments we fist need to enable a GitHub Webhook that will notify Brigade of a new release.
+
+Note: if using docker-for-desktop Kubernetes cluster follow [this guide](https://stefanprodan.com/2018/expose-kubernetes-services-over-http-with-ngrok/) to set up Ngrok tunneling to your local `brigade-github-app` service.
+
+- In your GitHub repository got to `Settings -> Webhooks -> Add Webhook`
+- In `Payload URL` enter your exposed brigade url e.g. http://e8432c17.ngrok.io/events/github
+- Change `Content type` to `application/json`
+- In `Secret` enter shared secret that was auto-generated during run of `brig project create` command above.
+- Choose `Let me select individual events.` unselect all options that have been preselected and choose `Branch or tag creation`. For the purpose of this tutorial we will be interested in tags created events only.
+
+## Service Implementation
+
+Our Python Products Microservice will expose REST Api that will handle adding products to a database.
+
+Service codebase has 3 main parts:
+
+**Service Code:**  
+`product` folder contains actual service code with product model defined in `models.py` 
+and service Api in `service.py`
+
+**Database Migrations:**  
+`migrations` folder contains Alembic database migrations scripts which will be executed 
+on each release.
+
+**Helm Charts:**  
+`charts` folder contains Helm chart for our service. Brigade script will use it to push 
+latest version of the service to our Preview Environments.
+
+To review full implementation details head over to the service repository: https://github.com/brigadecore/brigade-tutorial-app
+
+## Brigade Script
+
+When executing service deployment script we'll be fetching relevant git commit sha 
+for `prod` tag by calling GitHub REST API. To do so, we'll need to install additional 
+`node-fetch` dependency. We can do so by adding `brigade.json` file alongside our brigade 
+script. Brigade's `prestart` hook will load this file and install any dependencies listed there:
+
+```json
+{
+  "dependencies": {
+    "node-fetch": "^2.3.0"
+  }
+}
+```
+
+In brigade script implementation we will handle `create` events coming from 
+GtiHub gateway upon tag creation.  
+
+Script flow is as follows:
+
+- Grab a payload from incoming event
+
+```js
+events.on("create", event => {
+  const payload = JSON.parse(event.payload);
+  if (payload.ref_type !== "tag") {
+    console.log("skipping, not a tag commit");
+    return;
+  }
+  deployToEnvironments(payload).catch(error => {
+    throw error;
+  });
+});
+```
+- Use Kubernetes API to fetch all ConfigMaps from `brigade` namespace of type `preview-environment-config`
```

Comment:
Missing articles: `Use the Kubernetes API to fetch all ConfigMaps from the brigade namespace...`